src/battle_util.c

```diff
@@ -1679,17 +1679,12 @@ u8 DoFieldEndTurnEffects(void)
         switch (gBattleStruct->turnCountersTracker)
         {
         case ENDTURN_ORDER:
-            for (i = 0; i < gBattlersCount; i++)
-            {
-                gBattlerByTurnOrder[i] = i;
-            }
```

Comment:
why was this part removed?

Reply:
Only changes Turn Order but not actions, also Turn Order should already be correct from the actions taken during the turn.